Reproducible Failure in org.elasticsearch.index.mapper.flattened.FlattenedFieldSearchTests.testCardinalityAggregation

### CI Link

only saw it locally

### Repro line

./gradlew ':server:test' --tests "org.elasticsearch.index.mapper.flattened.FlattenedFieldSearchTests.testCardinalityAggregation" -Dtests.seed=6E5A7D9AA8064CD5 -Dtests.locale=sr -Dtests.timezone=America/Campo_Grande -Druntime.java=18

### Does it reproduce?

Yes

### Applicable branches

master

### Failure history

_No response_

### Failure excerpt

```
REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.index.mapper.flattened.FlattenedFieldSearchTests.testCardinalityAggregation" -Dtests.seed=EC78A8CA6F57105F -Dtests.locale=sv -Dtests.timezone=SystemV/MST7MDT -Druntime.java=18
  java.lang.AssertionError: 
    Expected: a value greater than <0L>
         but: <0L> was equal to <0L>
        at __randomizedtesting.SeedInfo.seed([EC78A8CA6F57105F:85146484AA481FC8]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.index.mapper.flattened.FlattenedFieldSearchTests.assertCardinality(FlattenedFieldSearchTests.java:275)
        at org.elasticsearch.index.mapper.flattened.FlattenedFieldSearchTests.testCardinalityAggregation(FlattenedFieldSearchTests.java:249)
```

Reproduces every time for me locally. Will mute.

--- a/server/src/test/java/org/elasticsearch/index/mapper/flattened/FlattenedFieldSearchTests.java
+++ b/server/src/test/java/org/elasticsearch/index/mapper/flattened/FlattenedFieldSearchTests.java
@@ -209,7 +209,6 @@ public class FlattenedFieldSearchTests extends ESSingleNodeTestCase {
         assertHitCount(searchResponse, 0L);
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/86115")
     public void testCardinalityAggregation() throws IOException {
         int numDocs = randomIntBetween(2, 100);
         int precisionThreshold = randomIntBetween(0, 1 << randomInt(20));

